lib/node_modules/@stdlib/console/log-each-map/docs/types/index.d.ts

```diff
@@ -0,0 +1,131 @@
+/*
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 4.1
+
+/**
+* Callback function that accepts no arguments.
+*/
+type NullaryCallback = () => any;
+
+/**
+* Callback function that accepts one argument.
+*/
+type UnaryCallback = ( x: any ) => any;
+
+/**
+* Callback function that accepts two arguments.
+*/
+type BinaryCallback = ( x: any, y: any ) => any;
+
+/**
+* Callback function that accepts three arguments.
+*/
+type TernaryCallback = ( x: any, y: any, z: any ) => any;
+
+/**
+* Callback function that accepts four arguments.
+*/
+type QuaternaryCallback = ( w: any, x: any, y: any, z: any ) => any;
+
+/**
+* Callback function that accepts five arguments.
+*/
+type QuinaryCallback = ( v: any, w: any, x: any, y: any, z: any ) => any;
+
+/**
+* Inserts array element values into a format string, applies a callback, and prints the result.
+*
+* @param str - Format string
+* @param clbk - Callback function
+* @param thisArg - Callback execution context
+*/
+declare function logEachMap( str: string, clbk: NullaryCallback, thisArg?: any ): void;
```

Comment:
These declarations can be significantly improved to better preserve type information.